SU-HSDO/suhumsci
# Stn 941  mvp mega menu

# READY FOR REVIEW

## Summary
Introduces styling for the We Mega Menu module. This styling is designed to apply only on the desktop view. The We Mega Menu is hidden on mobile, leaving the standard menu styling to govern the mobile menu. This module and styling does not impact sidebar menus or existing standard menus.

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. Run npm test

**Install the Module**
1. Navigate to Extend > Install New Module
2. Paste in the URL https://ftp.drupal.org/files/projects/we_megamenu-8.x-1.13.tar.gz
3. Navigate to Extend
4. Search or scroll down to find "Mega Menu for Drupal 8/9". If the box next to it is not checked, check it and click "Install" at the bottom of the window.

**Configure the Menu**
1. In Structure > Block layout, find the "Menu" region and select "Place Block"
2. Search for term "main"
3. Find the Main navigation block with a Category of "Drupal 8 Mega Menu" and select "Place Block"
4. Leave default configuration and select "Save block"
5. The mega menu will now appear on desktop only in the main menu region (below the standard menu)
6. To disable the standard menu on desktop, navigate to Appearance > Settings > Humsci Traditional
7. Scroll down to "Megamenu Settings" and check the box next to "Use Megamenu on Desktop". Save configuration.

- [x] Confirm level 1 styling of menu closely matches standard menu level 1 styling
- [x] Confirm level 1 active nav item has a bottom border matching the theme's secondary color
- [x] Confirm level 1 nav items have a gray bottom border on hover
- [x] Confirm the dropdown submenus are full-width
- [x] Confirm all list items are font family = Source Sans Pro, semibold 
- [x] Confirm level 2 **nolink** items are red (#8c1515), call caps, and not underlined
- [x] Confirm level 2 **link** items are red (#8c1515), all caps, and underlined
- [x] Confirm level 3 link items are black, not underlined
- [x] Confirm all level 2 and 3 link items have a gray-light (#f1f0ee) background on hover
- [x] Confirm the standard menu does not appear on desktop when the mega menu is present
- [x] Confirm the standard menu does appear on mobile

**Test on Dark Background**
1. Navigate to Appearance > Settings > Humsci Traditional
2. Under "Local Masthead Settings", select "Dark" 
3. Save configuration

- [x] Confirm that all level 1 nav items are white 
- [x] Confirm that level 1 active items have a gray bottom border (aligns with standard menu appearance on dark masthead)

**Test Admin Configuration**
1. Access the mega menu configuration at /admin/structure/we-mega-menu/main/config
2. Click any nav item that has a dropdown

- [x] Confirm the dropdown menu spans the width of the containing element so that columns can be visually arranged

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
